akka-stream-tests/src/test/scala/akka/stream/scaladsl/SourceSpec.scala

```diff
@@ -223,44 +225,46 @@ class SourceSpec extends AkkaSpec {
     val expected = List(9227465, 5702887, 3524578, 2178309, 1346269, 832040, 514229, 317811, 196418, 121393, 75025, 46368, 28657, 17711, 10946, 6765, 4181, 2584, 1597, 987, 610, 377, 233, 144, 89, 55, 34, 21, 13, 8, 5, 3, 2, 1, 1, 0)
 
     "generate a finite fibonacci sequence" in {
-      import GraphDSL.Implicits._
-
-      val source = Source.unfold((0, 1)) {
+      Source.unfold((0, 1)) {
         case (a, _) if a > 10000000 ⇒ None
         case (a, b)                 ⇒ Some((b, a + b) → a)
-      }
-      val result = Await.result(source.runFold(List.empty[Int]) { case (xs, x) ⇒ x :: xs }, 1.second)
-      result should ===(expected)
+      }.runFold(List.empty[Int]) { case (xs, x) ⇒ x :: xs }
+        .futureValue should be(expected)
     }
 
-    "generate a finite fibonacci sequence asynchronously" in {
-      import GraphDSL.Implicits._
-      import scala.concurrent.Future
-      import scala.concurrent.ExecutionContext.Implicits.global
+    "terminate with a failure if there is an exception thrown" in {
+      val t = new RuntimeException("expected")
+      whenReady(
+        Source.unfold((0, 1)) {
+          case (a, _) if a > 10000000 ⇒ throw t
+          case (a, b)                 ⇒ Some((b, a + b) → a)
+        }.runFold(List.empty[Int]) { case (xs, x) ⇒ x :: xs }.failed) {
+          _ should be(t)
+        }
+    }
 
-      val source = Source.unfoldAsync((0, 1)) {
+    "generate a finite fibonacci sequence asynchronously" in {
+      Source.unfoldAsync((0, 1)) {
         case (a, _) if a > 10000000 ⇒ Future.successful(None)
-        case (a, b)                 ⇒ Future(Some((b, a + b) → a))
-      }
-      val result = Await.result(source.runFold(List.empty[Int]) { case (xs, x) ⇒ x :: xs }, 1.second)
-      result should ===(expected)
+        case (a, b)                 ⇒ Future(Some((b, a + b) → a))(system.dispatcher)
+      }.runFold(List.empty[Int]) { case (xs, x) ⇒ x :: xs }
+        .futureValue should be(expected)
     }
 
-    "generate an infinite fibonacci sequence" in {
-      import GraphDSL.Implicits._
-
-      val source = Source.unfoldInf((0, 1)) {
-        case (a, b) ⇒ (b, a + b) → a
-      }
-      val result = Await.result(source.take(36).runFold(List.empty[Int]) { case (xs, x) ⇒ x :: xs }, 1.second)
-      result should ===(expected)
+    "generate an unbounded fibonacci sequence" in {
+      Source.unfoldInf((0, 1))({ case (a, b) ⇒ (b, a + b) → a })
+        .take(36)
+        .runFold(List.empty[Int]) { case (xs, x) ⇒ x :: xs }
+        .futureValue should be(expected)
```

Comment:
I have no idea, in any case, my suggestion on TLC for the test suite still stands. Lots of gardening required.